packages/@glimmer/runtime/lib/environment.ts

```diff
@@ -244,29 +247,76 @@ class TransactionImpl implements Transaction {
   }
 }
 
-export type ToBool = (value: unknown) => boolean;
+function defaultDelegateFn(delegateFn: Function | undefined, delegateDefault: Function) {
+  let defaultFn = delegateFn !== undefined ? delegateFn : delegateDefault;
 
-function toBool(value: unknown): boolean {
-  return !!value;
+  if (DEBUG) {
+    // Bind to `null` in DEBUG since these methods are assumed to be pure
+    // functions, so we can reassign them.
+    return defaultFn.bind(null);
```

Comment:
If this becomes an issue we could even wrap this in a proxy that asserts on `this` access, seems overkill at this point, but might eventually be useful.